modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostGISFeatureSourceOnlineTest.java

```diff
@@ -75,37 +75,12 @@ public void testBBOXOverlapsEncoding() throws Exception {
         assertTrue(areCRSEqual(CRS.decode("EPSG:4326"), bounds.getCoordinateReferenceSystem()));
     }
 
-    public void testEstimatedBounds() throws Exception {
```

Comment:
maybe overlooking something, but why remove testing with estimated bounds?

Reply:
Indeed must not be removed. I also don't see new tests for the fix.

Reply:
As far as I can tell, this test existed because estimated bounds was disabled by default in these tests for PostgreSQL (because "the tests assume non estimated extents", line 57 of PostGISTestSetup.java). But with this fix, disabling the feature for the tests to work is not required.

That means, now the testBounds test in the superclass is testing what was being tested by testEstimatedBounds, and the same for testEstimatedBoundsWithQuery (now achieved by testBoundsWithQuery).

That is also why you do not see new tests for the fix. The tests for this functionality already exist - they were just being bypassed before by setting setEstimatedExtentsEnabled to false.

Reply:
If you are not comfortable with that or not sure that my understanding is correct (it might not be, after all), it is possible to retain all the existing tests, keep setEstimatedExtentsEnabled false by default, and add a new test similar to testEstimatedBounds which turns it on and calls through to testGetFeaturesWithOffset to test the new functionality. That would be a smaller and simpler change.

Reply:
Yes please :-D

Reply:
Done!